test/dapp/main.js

```diff
@@ -3,46 +3,194 @@ async function start() {
 
   const web3 = new Web3(window.ethereum);
   console.log(web3);
-  const counterContract = new web3.eth.Contract(ContractInfo.abi, ContractInfo.address);
+  const counterContract = new web3.eth.Contract(
+    ContractInfo.abi,
+    ContractInfo.address
+  );
 
-  const increaseButton = document.querySelector('.increase-button');
-  increaseButton.addEventListener('click', async function () {
+  const increaseButton = document.querySelector(".increase-button");
+  increaseButton.addEventListener("click", async function () {
     await counterContract.methods.increase().send({ from: accounts[0] });
-    const txSent = document.createElement('div');
-    txSent.id = 'txSent';
+    const txSent = document.createElement("div");
+    txSent.id = "txSent";
     document.body.appendChild(txSent);
   });
 
-  const increaseFeesButton = document.querySelector('.increase-fees-button');
-  increaseFeesButton.addEventListener('click', async function () {
+  const increaseFeesButton = document.querySelector(".increase-fees-button");
+  increaseFeesButton.addEventListener("click", async function () {
     await counterContract.methods.increase().send({ from: accounts[0] });
-    const txSent = document.createElement('div');
-    txSent.id = 'feesTxSent';
+    const txSent = document.createElement("div");
+    txSent.id = "feesTxSent";
     document.body.appendChild(txSent);
   });
 
-  const connectButton = document.querySelector('.connect-button');
-  connectButton.addEventListener('click', async function () {
+  const connectButton = document.querySelector(".connect-button");
+  connectButton.addEventListener("click", async function () {
     accounts = await web3.eth.requestAccounts();
-    const connected = document.createElement('div');
-    connected.id = 'connected';
+    const connected = document.createElement("div");
+    connected.id = "connected";
     document.body.appendChild(connected);
   });
 
-  const signButton = document.querySelector('.sign-button');
-  signButton.addEventListener('click', async function () {
-    const message = web3.utils.sha3('TEST');
+  const signButton = document.querySelector(".sign-button");
+  signButton.addEventListener("click", async function () {
+    const message = web3.utils.sha3("TEST");
     await web3.eth.sign(message, accounts[0]);
-    const signed = document.createElement('div');
-    signed.id = 'signed';
+    const signed = document.createElement("div");
+    signed.id = "signed";
     document.body.appendChild(signed);
   });
 
-  const transferButton = document.querySelector('.transfer-button');
-  transferButton.addEventListener('click', async function () {
-    await web3.eth.sendTransaction({ to: accounts[0], from: accounts[0], value: web3.utils.toWei('0.01') });
-    const transfer = document.createElement('div');
-    transfer.id = 'transferred';
+  const signTypedDataButton = document.querySelector(".sign-typedData-button");
```

Comment:
do you thing we could start writing tests without adding code to test dapp like:
```typescript
const resultPromise  = testPage.evaluate(async function() {
   //prepare data
   return window.ethereum.sendAsync(
      {
        method,
        params,
        from: accounts[0],
      });
});
await dappteer.signTypedData()
const result = await resultPromise
```
?

Reply:
This is definitely an idea worth trying. Adding some buttons to click feels a little clunky for such automated tests indeed :)

Reply:
aand Bero did it before me :)